mlhaiti/accounts/api/views.py

```diff
@@ -33,9 +37,86 @@ def create(self, request, *args, **kwargs):
             {
                 'status':400,
                 'error':True,
-                'data':[],
+                'data':{},
                 'message':serializer.errors
             },
             status=status.HTTP_400_BAD_REQUEST
         )
 
+class UserLoginView(views.APIView):
+    """
+        User Login
+    """    
+    permission_classes = (permissions.AllowAny,)
+    def post(self,request):
+        username = request.data.get('username')
+        password = request.data.get('password')
+        if username and password:
+            user = authenticate(username=username, password=password)
+            if user:
+                token = Token.objects.get_or_create(user=user)[0]
+                data = UserSerializer(user).data
+                data['token'] = token.key 
+                return Response(
+                    {
+                        'error':False,
+                        "message": _("User Login Successfully"),
+                        "data":data,
+                        "status": 200
+                    },
+                    status=status.HTTP_200_OK
+                )
+            message = _("Unable to login with given credentials")
+            return Response(
+                {
+                    'error':True,
+                    "error_message": message ,
+                    'data': {},
+                    'status':401
+                },
+                status=status.HTTP_401_UNAUTHORIZED
+            )
+        message = _("Invalid Credentials")
+        return Response(
+            {
+                'error':True, 
+                "error_message": message ,
+                'data': {},
+                'status':401
+            },
+            status=status.HTTP_401_UNAUTHORIZED
+        )
+
+
+class UserLogoutView(views.APIView):
+    """
+        User Logout
+    """    
+    def post(self, request):
+        return self.logout(request)
+
+    def logout(self, request):
+        try:
+            request.user.auth_token.delete() # delete the user authenticated user token
+        except (AttributeError, ObjectDoesNotExist):
+            return Response(
+                {
+                    'error':True,
+                    "message": _("No user token found"),
+                    "data":{},
+                    "status": 401
+                },
+                status=status.HTTP_401_UNAUTHORIZED
+            )
+
+        logout(request) # Delete session
+        return Response(
+            {
+                'error':False,
+                "message": _("Successfully logged out."),
+                "data":{},
+                "status": 200
+            },
+            status=status.HTTP_200_OK
+        )
```

Comment:
Li pa ranvoye mesaj erè ou mandel pou ranvoye a loske ou pa pase token nan nan header a. li jis afiche saa:

{
      detail: "Authentication credentials were not provided."
}

Reply:
Loske mwen vle logout user a mwen rankontre menm pwoblem lan. Menm mesaj la

Reply:
Login nan mache byen. 👍🏾

Reply:
Yes, paske li se yon endpoint proteje.  Ou dwe pase token nan, nan header, si n pap fe logout ak token, it's useless to have a logout endpoint